### PR TITLE
Modified run_info to add searchable attributes decorator and made a test

### PR DESCRIPTION
**Pull request by alexacole**
Signed-off-by: Alexa Cole <alexa.cole@ufl.edu>

## Related Issues/PRs

#6867

## What changes are proposed in this pull request?

Enable users to search for runs by run ID.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/utils/test_search_utils.py [hunk 1]
```diff
@@ -577,3 +579,46 @@ def test_pagination(page_token, max_results, matching_runs, expected_next_page_t
...
+            status=RunStatus.to_string(RunStatus.FAILED),
+            start_time=0,
+            end_time=1,
+            lifecycle_stage=LifecycleStage.ACTIVE,
+        ),
+            run_data=RunData([], [], [])),
+    ]
+
+    run_id = "1"
+    returned_id = runs[1].RunInfo.run_id
```
Comment (BenWilson2) [inline tests/utils/test_search_utils.py:623]: Did you want to test using `SearchUtils.filter(runs, "attributes.run_id = 1")` and ensure that the return is just that run_id?
I think lines 131 & 495 will need to be modified as well.

File: mlflow/entities/run_info.py [hunk 2]
```diff
@@ -92,6 +92,7 @@ def run_uuid(self):
         return self._run_uuid
 
     @property
+    @searchable_attribute
```
Comment (harupy) [inline mlflow/entities/run_info.py:95]:
```suggestion
    @searchable_attribute
```

File: tests/utils/test_search_utils.py [hunk 3]
```diff
@@ -577,3 +579,46 @@ def test_pagination(page_token, max_results, matching_runs, expected_next_page_t
 def test_invalid_page_tokens(page_token, error_message):
     with pytest.raises(MlflowException, match=error_message):
         SearchUtils.paginate([], page_token, 1)
+
+
+
+def test_single_run_by_ID(run_id, returned_id):
```
Comment (harupy) [inline tests/utils/test_search_utils.py:585]: Let's remove this and add tests in `tests/store/tracking/test_file_store.py` and `tests/store/tracking/test_sqlalchemy_store.py`

File: tests/utils/test_search_utils.py (hunk 1 above)
Comment (alexacole) [reply tests/utils/test_search_utils.py:623]: Thank you. That's what I wanted to test.

**Comment by harupy** [comment]
@alexacole Any updates here?

**Review by harupy — APPROVED** [review]
@alexacole LGTM! Thanks for the contribution :)

**Pull request merged**